internal/wasm/memory.go

```diff
@@ -39,6 +49,14 @@ type MemoryInstance struct {
 	Shared        bool
 	// definition is known at compile time.
 	definition api.MemoryDefinition
+
+	// Mux is used in interpreter mode to prevent overlapping calls to atomic instructions,
+	// introduced with WebAssembly threads proposal.
+	Mux sync.RWMutex
```

Comment:
You seem to be using this as a `sync.Mutex` only, which I don't disagree with (unless it's *proven* that we have many concurrent readers, few writers *and* this helps, a plain `Mutex` is probably cheaper).

If this becomes a bottleneck, intuition tells me having (e.g.) `[4]sync.Mutex` and using (e.g.) bits 5 and 6 of the address to index that array would be better than a `RWMutex`.

Reply:
Ah thanks, for now I'll switch to normal mutex, I just resurrected this which used to be used with Grow without noticing